fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -547,10 +547,22 @@ private ScoredPlan planFilter(@Nonnull PlanContext planContext, @Nonnull QueryCo
         return withInJoin;
     }
 
-    @Nullable
-    private ScoredPlan planFilterWithInJoin(@Nonnull PlanContext planContext, @Nonnull InExtractor inExtractor, boolean needOrdering) {
-        int maxNumReplans = Math.max(getConfiguration().getMaxNumReplansForInToJoin(), 0);
-        boolean allowNonSargedInBindings = getConfiguration().getMaxNumReplansForInToJoin() < 0;
+    private static final class PlanWithInExtractor {
+        @Nonnull
+        private final ScoredPlan plan;
+        @Nonnull
+        private final InExtractor inExtractor;
+
+        PlanWithInExtractor(@Nonnull ScoredPlan plan, @Nonnull InExtractor inExtractor) {
+            this.plan = plan;
+            this.inExtractor = inExtractor;
+        }
+    }
+
+    private PlanWithInExtractor planFilterForInWithReplans(@Nonnull PlanContext planContext, @Nonnull InExtractor inExtractor, boolean needOrdering, int maxNumReplansInConfig) {
+        int maxNumReplans = Math.max(maxNumReplansInConfig, 0);
+        boolean allowNonSargedInBindings = maxNumReplansInConfig < 0;
+
```

Comment:
Similarly, can we rename `planFilterForInJoin` to something like `planFilterForIn` or something more fundamental like `planBaseAccessPath` as that method is the lowest of the filter descenders IIRC.

Reply:
I went with `planExtractedInsFilter` because I think the most relevant thing for the user is to note that the `filter` being planned is different from the original query filter. It's not quite `planBaseAccessPath` because it applies some compensation on top of base scans (e.g., it creates intersections and it applies filters)